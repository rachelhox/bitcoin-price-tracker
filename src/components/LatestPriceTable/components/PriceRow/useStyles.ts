import { css, useTheme } from '@mui/material';

const useStyles = () => {
  const theme = useTheme();
  return {
    row: css({
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      [theme.breakpoints.up('laptop')]: {
        width: '100%',
      },
    }),
    h6: css({
      color: theme.palette.primary.main,
      fontSize: theme.spacing(2.5),
      [theme.breakpoints.up('laptop')]: {
        fontSize: theme.spacing(4),
      },
    }),
    price: css({
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'space-between',
      alignItems: 'flex-end',
    }),
    body1: css({
      color: theme.palette.primary.main,
      fontSize: theme.spacing(2),
      [theme.breakpoints.up('laptop')]: {
        fontSize: theme.spacing(3),
      },
    }),
    subtitle1: css({
      fontSize: theme.spacing(1),
      color: theme.palette.custom.gray,
      [theme.breakpoints.up('laptop')]: {
        fontSize: theme.spacing(1.5),
      },
    }),
  };
};

export default useStyles;
