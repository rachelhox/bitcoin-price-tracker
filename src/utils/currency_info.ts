interface IStringIndex extends Record<string, any> {}

/**
 * Takes the key name and returns the location. Will return null if undefined
 */

const currencies: IStringIndex = {
  usd: {
    name: 'US Dollar',
    key: 'usd',
    unit: '$',
    type: 'fiat',
  },
  aed: {
    name: 'United Arab Emirates Dirham',
    key: 'aed',
    unit: 'DH',
    type: 'fiat',
  },
  eur: {
    name: 'Euro',
    key: 'eur',
    unit: '€',
    type: 'fiat',
  },
  aud: {
    name: 'Australian Dollar',
    key: 'aud',
    unit: 'A$',
    type: 'fiat',
  },
  cad: {
    name: 'Canadian Dollar',
    key: 'cad',
    unit: 'CA$',
    type: 'fiat',
  },
  chf: {
    name: 'Swiss Franc',
    key: 'chf',
    unit: 'Fr.',
    type: 'fiat',
  },
  gbp: {
    name: 'British Pound Sterling',
    key: 'gbp',
    unit: '£',
    type: 'fiat',
  },
  hkd: {
    name: 'Hong Kong Dollar',
    key: 'hkd',
    unit: 'HK$',
    type: 'fiat',
  },
  sgd: {
    name: 'Singapore Dollar',
    key: 'sgd',
    unit: 'S$',
    type: 'fiat',
  },
  cny: {
    name: 'Chinese Yuan',
    key: 'cny',
    unit: '¥',
    type: 'fiat',
  },
  jpy: {
    name: 'Japanese Yen',
    key: 'jpy',
    unit: '¥',
    type: 'fiat',
  },
  inr: {
    name: 'Indian Rupee',
    key: 'inr',
    unit: '₹',
    type: 'fiat',
  },
  krw: {
    name: 'South Korean Won',
    key: 'krw',
    unit: '₩',
    type: 'fiat',
  },
};

export const getCurrencyInfo = (key: string | number) => {
  return currencies[key] ?? {};
};
