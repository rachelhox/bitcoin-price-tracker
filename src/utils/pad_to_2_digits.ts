/* eslint-disable */

export const padTo2Digits = (num: any) => {
  return String(num).padStart(2, '0');
};
