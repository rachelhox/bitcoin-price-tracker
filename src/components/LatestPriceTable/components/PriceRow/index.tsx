/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
import { Box, Typography, useTheme } from '@mui/material';
import { convertToMoney } from '@utils';
import useStyles from './useStyles';

interface PriceRowProp {
  priceTableData: CurrencyValueProp[];
}

const PriceRow = ({ priceTableData }: PriceRowProp) => {
  const styles = useStyles();
  return (
    <>
      {priceTableData.map((price) => (
        <Box css={styles.row} key={price.key}>
          <Typography variant="h6" css={styles.h6}>
            BTC/{price.key.toUpperCase()}
          </Typography>
          <Box css={styles.price}>
            <Typography variant="body1" css={styles.body1}>
              {convertToMoney(price.value)}
            </Typography>
            <Typography variant="subtitle1" css={styles.subtitle1}>
              {`${price.unit} ${convertToMoney(price.value)}`}
            </Typography>
          </Box>
        </Box>
      ))}
    </>
  );
};

export default PriceRow;
