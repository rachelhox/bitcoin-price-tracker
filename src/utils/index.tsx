import { getCurrencyInfo } from './currency_info';
import { convertToMoney } from './convert_to_money';
import { padTo2Digits } from './pad_to_2_digits';

export { getCurrencyInfo, convertToMoney, padTo2Digits };
