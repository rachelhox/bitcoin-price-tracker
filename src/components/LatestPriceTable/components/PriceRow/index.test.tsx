import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import PriceRow from '.';

// ==================================
// mocks
// ==================================

const mockPriceTableData = [
  {
    key: 'aed',
    name: 'United Arab Emirates Dirham',
    type: 'fiat',
    unit: 'DH',
    value: 63890.518,
  },
  {
    key: 'usd',
    name: 'US Dollar',
    type: 'fiat',
    unit: '$',
    value: 17394.665,
  },
];

// ==================================
// unit tests
// ==================================

describe('component: PriceRow', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <PriceRow priceTableData={mockPriceTableData} />
      </MockTheme>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
