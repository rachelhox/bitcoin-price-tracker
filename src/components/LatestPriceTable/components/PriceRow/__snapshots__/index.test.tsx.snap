// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: PriceRow matches snapshot 1`] = `
[
  <div
    className="MuiBox-root css-dqueu"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-1dgyoj8-MuiTypography-root"
    >
      BTC/
      AED
    </h6>
    <div
      className="MuiBox-root css-6lnlif"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-detvd-MuiTypography-root"
      >
        63,891
      </p>
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 css-zigcs7-MuiTypography-root"
      >
        DH 63,891
      </h6>
    </div>
  </div>,
  <div
    className="MuiBox-root css-dqueu"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-1dgyoj8-MuiTypography-root"
    >
      BTC/
      USD
    </h6>
    <div
      className="MuiBox-root css-6lnlif"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-detvd-MuiTypography-root"
      >
        17,395
      </p>
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 css-zigcs7-MuiTypography-root"
      >
        $ 17,395
      </h6>
    </div>
  </div>,
]
`;
